source/firstprogram.rst

```diff
@@ -1,402 +1,621 @@
 
-Any model built using this framework contains different classes which are essentially extensions of different `Nodes <#>`_. To properly build a SIR model from scratch, you will first need to define these classes and the properties associated with them. These classes are grouped under different components of the model. Here, the SIR model has two different components:-
-1. `Agent <#>`_
-2. `Network <#>`_
+Any model built on this framework contains various classes which are essentially extensions of different Nodes. To build a SIR model from scratch, one needs to define the said classes and the properties and relationships associated with them. In the basic SIR there are two main components:
 
-These components are related to each other via simple relations which are described in detail later.
+1. `Agent`
+2. `Network`
 
-Setting up the components of the Network
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Agents are simply the individuals in the population and the Network is the manner in each of these individuals interact with one another. 
+However, to begin writing this program we assume that all the individuals are present in one location, hence neglecting the Network component. 
 
-The Network class gives a sense of geography to the model. The different components of the Network are simple locations like houses, offices, schools etc. The agents move about between these locations as shown in the figure below.
+Creating an Empty Class:
+^^^^^^^^^^^^^^^^^^^^^^^^
+Before we get to the actual coding aspect of the code, it is essential that we set up an empty and make sure that the framework is not giving any errors. 
+Navigate to ``BharatSim/src/main/scala/com/bharatsim/examples/epidemiology`` in IntelliJ or VSCode and right click on the folder. Upon right clicking, select the New Package option and name it SIR_Scratch. An empty folder should appear named SIR_Scratch and create a new scala class called Main. A yellow circular icon should appear. 
 
-The first step to building your model is to create locations which are part of the Network, by creating a new scala `class <https://docs.scala-lang.org/tour/classes.htm>`_. You can create a new class for adding houses to the network, in the following way:-
+The empty class should look like this. 
 
-1. Since ``House`` is a component of the Network, you have to import the Network class.
+.. image:: _static/images/New_class.png
+
+Now we can define a main function that has no input and has no output. The syntax and indentation of defining a function is as follows
 
 .. code-block:: scala
 
-  import com.bharatsim.engine.models.Network
+    def main(args: Array[String]): Unit = {
+    }
+
+The args means that the argument or the input is an array of Strings and the output is of type Unit, which corresponds to void means that there is no output. The code should look like this, 

```

Comment:
Put args in "backticks" (i.e. the \` symbol) so that it looks like this: `args`.

Reply:
Same with `String`, and `Unit`. Also, "corresponds to void" won't make any sense to someone who hasn't worked on programming languages before.